src/update_halo.jl

```diff
@@ -117,14 +117,14 @@ let
     end
 
 
-    # (CUDA functions)
+    # (CUDA, AMDGPU functions)
 
     function free_gpubufs(bufs)
         if (bufs !== nothing)
             for i = 1:length(bufs)
                 for n = 1:length(bufs[i])
                     if is_cuarray(bufs[i][n])  CUDA.unsafe_free!(bufs[i][n]); bufs[i][n] = []; end
-                    if is_rocarray(bufs[i][n]) AMDGPU.unsafe_free!(bufs[i][n]); bufs[i][n] = []; end
+                    # if is_rocarray(bufs[i][n]) AMDGPU.unsafe_free!(bufs[i][n]); bufs[i][n] = []; end # DEBUG: unsafe_free should be managed in AMDGPU
```

Comment:
I don't follow why you commented this...?

Reply:
In AMDGPU, no `unsafe_free` is not needed as it should be managed (unlike CUDA)